src/microbe_stage/systems/MucocystSystem.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace Systems;
+
+using System;
+using Components;
+using DefaultEcs;
+using DefaultEcs.System;
+using DefaultEcs.Threading;
+
+/// <summary>
+///   Handles gaining immortality from mucocyst ability
+/// </summary>
+[With(typeof(MicrobeControl))]
+[With(typeof(Health))]
+[With(typeof(CompoundStorage))]
+[Without(typeof(AttachedToEntity))]
+[WritesToComponent(typeof(CellProperties))]
+[RunsAfter(typeof(OrganelleComponentFetchSystem))]
+[RunsBefore(typeof(MicrobeMovementSystem))]
+[RuntimeCost(7)]
```

Comment:
This seems a bit suspicious to me, did you profile the system and confirm it takes up 7x more time than just simple systems?

Reply:
I'm still not proffesional at systems, how much would you give here?

Reply:
I would need to profile but I would *guess* that the performance impact of this system would be at most a 2 but most likely just 1 (so just removing this attribute) is the closest.